### PR TITLE
Add support for OAuth and Payment Initiation

### PR DESCRIPTION
**Pull request by jacksiegel**
Adds support for the `oauthNonce`, `oauthRedirectUri`, and `oauthStateId` Link configuration parameters required for OAuth authentication flows when initializing Link with European country codes, as well as support for the Payment Initiation product.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,8 +49,7 @@ a more holistic understanding of the various Link options.
 <PlaidLink
   clientName="Your app name"
   env="sandbox"
-  institution={null}
-  publicKey={PLAID_PUBLIC_KEY}
+  key={PLAID_PUBLIC_KEY}
```
Comment (skansal-plaid) [inline README.md]: We also need to modify these spots to do `s/publicKey/key`: https://github.com/plaid/react-plaid-link/search?q=publicKey&unscoped_q=publicKey

**Comment by skansal-plaid** [comment]
Also did running `make setup; make start` and then visiting localhost:3000 work for you? It does not work for me locally.

File: README.md (hunk 1 above)
Comment (skansal-plaid) [reply README.md]: https://github.com/plaid/react-plaid-link/pull/72

Comment (jacksiegel) [reply README.md]: Good catch – I got mixed up by the `Plaid.create` call. Better to have a 1:1 mapping from the React props to the Link configuration parameter names in our docs.

**Comment by jacksiegel** [comment]
@skansal-plaid it does not work for me.

**Comment by skansal-plaid** [comment]
@jacksiegel Thank you for confirming, I have filed DEVEXP-29 for this.

**Pull request merged**